extendr-api/src/robj/tests.rs

```diff
@@ -153,10 +153,26 @@ fn test_try_from_robj() {
 
         assert_eq!(<Vec::<i32>>::try_from(Robj::from(1)), Ok(vec![1]));
         assert_eq!(<Vec::<f64>>::try_from(Robj::from(1.)), Ok(vec![1.]));
+        assert_eq!(<Vec::<Bool>>::try_from(Robj::from(TRUE)), Ok(vec![TRUE]));
+        assert_eq!(<Vec::<u8>>::try_from(Robj::from(0_u8)), Ok(vec![0_u8]));
+
+        assert_eq!(<&[i32]>::try_from(Robj::from(1)), Ok(&[1][..]));
+        assert_eq!(<&[f64]>::try_from(Robj::from(1.)), Ok(&[1.][..]));
+        assert_eq!(<&[Bool]>::try_from(Robj::from(TRUE)), Ok(&[TRUE][..]));
+        assert_eq!(<&[u8]>::try_from(Robj::from(0_u8)), Ok(&[0_u8][..]));
+
+        // Note the Vec<> cases use the same logic as the slices.
+        assert_eq!(<&[i32]>::try_from(Robj::from(1.0)), Err(Error::ExpectedInteger(r!(1.0))));
+        assert_eq!(<&[f64]>::try_from(Robj::from(1)), Err(Error::ExpectedReal(r!(1))));
```

Comment:
Same as above. Shouldn't this succeed?

Reply:
Ditto.

Reply:
The slices are a bit of an exception as they give immutable reference access to the interiors of the R objects.

You can still check for NA using `value.is_na()` but checking all the values of a gigabyte-long vector may be undesirable.

Reply:
We should consider having mutable slices also `&mut [i32]` etc.

This would allow the modification of R objects in-place.

Reply:
Ok, I get the argument for the slices, but not for vectors. There we're making a copy of the data, correct? So then we should use the same conversion and rules that we're using for scalars (convert 1.0 to 1 etc.)

Reply:
> The slices are a bit of an exception as they give immutable reference access to the interiors of the R objects.

So, this is also the reason why no other int/float variants (e.g. i64 and f32) are supported in slice's case, right? I'll document the rule on #244.